CDCgov/prime-reportstream
# 14721: verify action params and SQL param escaping

This PR adds some data sanitization and validation around the sender ip and external name fields that get in the action log and report file tables respectively.

Test Steps:
There is decent test coverage, but the external name can be tested

1. Send a report with the `payloadname` header with a value ~3000 characters long
2. Make sure the report is not accepted

## Changes
- Validates that the sender IP is a real IP before inserting it
- Errors the report if the payloadname is larger the external name db column
- Switches to use jackson for some of the action params
- Adds some more tests

## Checklist

### Testing

### Process

## Linked Issues
- #14721 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
